examples/seq2seq/run_seq2seq.py

```diff
@@ -386,7 +386,7 @@ def main():
 
     # For translation we set the codes of our source and target languages (only useful for mBART, the others will
     # ignore those attributes).
-    if data_args.task.startswith("translation"):
+    if data_args.task.startswith("translation") or isinstance(tokenizer, (MBartTokenizer, MBartTokenizerFast)):
```

Comment:
LGMT, thanks a lot for your PR!

@stas00  @sgugger  think we can remove the check for task and tokenizer, and just set the src and tgt lang if it's provided, the rest of the tokenizers will ignore these attributes anyway.

Reply:
Otherwise, we will end up checking for each different multilingual tokenizer, for example, we just  added `MBart50Tokenizer` which also needs these attributes, and then there's `M2M100` coming soon, which also needs src and tgt lang

Reply:
Good point - but I don't think it's a good idea to set it if it's passed, because we should restrict this only to the tokenizers that need them as discussed here to avoid ambiguity https://github.com/huggingface/transformers/issues/10164

Why not then check if the tokenizer has these attributes and only then set them then - and assert if it wasn't provided? It'd work for any tokenizer then.

Reply:
That definitely sounds better! Right now `MBartTokenizer` does not set default or  `None` values for `src_lang` and `tgt_lang`, so `hasttar` will fail, but we can add those in the init of `MBartTokenizer`, which shouldn't break anything.

Reply:
No, they don't have it. So my idea won't work with the way things are now.

But if you look at this:

https://github.com/huggingface/transformers/blob/1c8c2d9ab34b8c8d326db9e0608f8e54cfccb885/src/transformers/models/mbart/tokenization_mbart_fast.py#L174-L185

it's not even designed properly for that - and sets these on the fly during `prepare_seq2seq_batch` (with hardcoded high percentage chance incorrect fallback values - ouch). So the current API is very error-prone.

Well, perhaps the simplest solution is to change these tokenizer classes to have the attributes that they require and then we can test if they have these attributes? That is other tokenizer class won't have these attributes and my idea would work.

Reply:
Ah, we proposed the same thing - it just took me a while to complete my comment so I didn't see yours. We agree then.

Reply:
`MBart50TokenizerFast` as well, right?

Reply:
I agree with the proposed change. In the meantime, this quick fix can be merged (and reverted with the work of MBart tokenziers). Does that sound right to you?

Reply:
Sound good to me!